### PR TITLE
cpupower.py: cpufreq feature only supported on PowerNV

### PR DESCRIPTION
**Pull request by shirishaganta**
In PowerVM cpufreq is not supported
so changed the script that cpufreq supports only in PowerNV environment

Signed-off-by: Shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
without patch in SLES15-SP1:

JOB ID     : 12c3046ddfa3fbece8036a9ad332434e66e33a4f
JOB LOG    : /root/avocado/avocado-fvt-wrapper/results/job-2019-06-14T13.53-12c3046/job.log
 (1/1) cpupower.py:cpupower.test: CANCEL: sysfs directory for cpufreq is unavailable. (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 0.39 s
JOB HTML   : /root/avocado/avocado-fvt-wrapper/results/job-2019-06-14T13.53-12c3046/results.html

with patch in SLES15-SP1:

JOB ID     : 6e42835551bdba1a4f90f583bcd5438791b85902
JOB LOG    : /root/avocado/avocado-fvt-wrapper/results/job-2019-06-14T14.17-6e42835/job.log
 (1/1) cpupower.py:cpupower.test: CANCEL: cpufreq is supported only on PowerNV machines (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 0.39 s
JOB HTML   : /root/avocado/avocado-fvt-wrapper/results/job-2019-06-14T14.17-6e42835/results.html

File: cpu/cpupower.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class cpupower(Test):
 
     def setUp(self):
         if not os.path.exists('/sys/devices/system/cpu/cpu0/cpufreq'):
-            self.cancel('sysfs directory for cpufreq is unavailable.')
+            self.cancel('cpufreq is supported only on PowerNV machines')
```
Comment (narasimhan-v) [inline cpu/cpupower.py]: The check is for directory existence, and the cancel message is saying about PowerNV.
Change the condition also please.

File: cpu/cpupower.py [hunk 2]
```diff
@@ -32,8 +31,9 @@ class cpupower(Test):
     """
 
     def setUp(self):
-        if not os.path.exists('/sys/devices/system/cpu/cpu0/cpufreq'):
-            self.cancel('sysfs directory for cpufreq is unavailable.')
+        if not 'PowerNV' in cpu._get_cpu_info():
```
Comment (harish-24) [inline cpu/cpupower.py]: Suggestion: Better to add a similar implementation of https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/ras/ras.py#L58

File: cpu/em_cpufreq.py [hunk 3]
```diff
@@ -33,8 +32,9 @@ def setUp(self):
         """
         Verify the system is Baremetal and cpupower tool is installed.
         """
-        if not os.path.exists('/sys/devices/system/cpu/cpu0/cpufreq'):
-            self.cancel('CPUFREQ is supported only on Power NV')
+        if not 'PowerNV' in cpu._get_cpu_info():
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: here too

File: cpu/cpupower.py (hunk 2 above)
Comment (PraveenPenguin) [reply cpu/cpupower.py]: i think should be ok as in none of any combination on this test should run on any other arch

File: cpu/cpupower.py [hunk 4]
```diff
@@ -31,9 +33,9 @@ class cpupower(Test):
     :avocado: tags=cpu,power,privileged
     """
 
+    @skipIf(not IS_POWER_NV, "This test is not supported on  PowerVM platform")
```
Comment (harish-24) [inline cpu/cpupower.py]: extra space here: `supported on  PowerVM`

File: cpu/cpupower.py [hunk 5]
```diff
@@ -31,9 +33,9 @@ class cpupower(Test):
...
     def setUp(self):
-        if not os.path.exists('/sys/devices/system/cpu/cpu0/cpufreq'):
-            self.cancel('sysfs directory for cpufreq is unavailable.')
+
```
Comment (harish-24) [inline cpu/cpupower.py]: Better to add some Docstring or least remove the empty line.

File: cpu/cpupower.py (hunk 4 above)
Comment (shirishaganta) [reply cpu/cpupower.py]: yeah i will remove that

File: cpu/cpupower.py (hunk 5 above)
Comment (shirishaganta) [reply cpu/cpupower.py]: sure

**Pull request merged**